app/Models/Course.php

```diff
@@ -18,14 +18,72 @@ class Course extends Model
         'description'
     ];
 
+    public function scopeSearch($query, $data)
+    {
+        if (isset($data['course_name'])) {
+            $query->where('name', 'LIKE', "%". $data['course_name'] ."%");
+        }
+
+        if (isset($data['course_create'])) {
+            $data['course_create'] == "newest" ? $query->orderby('courses.created_at', 'DESC') : $query->orderby('courses.created_at');
+        }
+
+        if (isset($data['teacher_id']) && !empty($data['teacher_id'])) {
+            $query->join('teacher_course', 'courses.id', '=', 'teacher_course.course_id')
+                ->where('user_id', $data['teacher_id']);
+        }
+
+        if (isset($data['order_by_learner']) && !empty($data['order_by_learner'])) {
+            $query->withCount('users');
+            $query->orderBy('users_count', $data['order_by_learner']);
+        }
+
+        if (isset($data['order_by_time']) && !empty($data['order_by_time'])) {
+            $query->withCount('lessons')->withSum('lessons', 'times');
+            $query->orderBy('lessons_sum_times', $data['order_by_time']);
+        }
+
+        if (isset($data['order_by_lesson']) && !empty($data['order_by_lesson'])) {
+            $query->withCount('lessons');
```

Comment:
câu query này hiểu thế nào nhỉ?

Reply:
select `courses`.*, 
(select count(*) from `lessons` where `courses`.`id` = `lessons`.`course_id` and `lessons`.`deleted_at` is null) as `lessons_count` 
from `courses`